### PR TITLE
BuDDI packaging

### PR DESCRIPTION
**Pull request by nrosed**
In this PR I mostly fixed bugs with the packaging of BuDDI, but also some sc_preprocessing bugs: 

1. Removed files from this repo that should be in the buddi_analysis repo
    a. `buddi/sc_preprocessing/2_make_pseudobulks/*.ipynb`
    b. `buddi/sc_preprocessing/0_qc_notebooks/*.ipynb`
2. Fixed paths and version pins for appropriate packaging
    a. `buddi/buddi.py`
    b. `buddi/comparator_models/cvae.py`
    c. `buddi/comparator_models/vae.py`
    d. `setup.py`
3. Fixed a bug where some cell type proportions were 0 for a cell type. They are now guaranteed to be >0
    a. `buddi/preprocessing/sc_preprocess.py`

### CODE REVIEW

File: .gitignore
Comment (ivichadriana) [inline .gitignore:1]: Do you want this file?

File: buddi/buddi.py
Comment (ivichadriana) [inline buddi/buddi.py:1]: just a couple paths changed, all good

File: buddi/comparator_models/cvae.py
Comment (ivichadriana) [inline buddi/comparator_models/cvae.py:1]: just a couple paths changed, all good

File: buddi/comparator_models/vae.py
Comment (ivichadriana) [inline buddi/comparator_models/vae.py:1]: just a couple paths changed, all good

File: buddi/preprocessing/sc_preprocess.py [hunk 1]
```diff
@@ -52,11 +50,11 @@ def get_cell_type_sum(in_adata, cell_type_id, num_samples):
...
 
 # method to generate a proportion vector
-def gen_prop_vec_lognormal(len_vector, num_cells):
+def gen_prop_vec_lognormal(len_vector, num_cells): # pseudo
 
   rand_vec = np.random.lognormal(5, np.random.uniform(1,3), len_vector) # 1
-
-  rand_vec = np.round((rand_vec/np.sum(rand_vec))*num_cells)
+  rand_vec = rand_vec + 1
+  rand_vec = np.ceil((rand_vec/np.sum(rand_vec))*num_cells)
```
Comment (ivichadriana) [inline buddi/preprocessing/sc_preprocess.py:57]: seems to have done the trick! I also noticed the 0 in the proportions when i used this function. after the rounding, some cells were 0. glad it's fixed:D

File: buddi/sc_preprocessing/0_qc_notebooks/scanpy_analysis_kang.ipynb
Comment (ivichadriana) [inline buddi/sc_preprocessing/0_qc_notebooks/scanpy_analysis_kang.ipynb:1]: just deleted, all good.

File: buddi/sc_preprocessing/0_qc_notebooks/scanpy_analysis_liver.ipynb
Comment (ivichadriana) [inline buddi/sc_preprocessing/0_qc_notebooks/scanpy_analysis_liver.ipynb:1]: just deleted, all good.

File: buddi/sc_preprocessing/0_qc_notebooks/scanpy_analysis_liver_DE.ipynb
Comment (ivichadriana) [inline buddi/sc_preprocessing/0_qc_notebooks/scanpy_analysis_liver_DE.ipynb:1]: just deleted, all good.

File: buddi/sc_preprocessing/0_qc_notebooks/scanpy_analysis_liver_bulk.ipynb
Comment (ivichadriana) [inline buddi/sc_preprocessing/0_qc_notebooks/scanpy_analysis_liver_bulk.ipynb:1]: just deleted, all good.

File: buddi/sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-kang.ipynb
Comment (ivichadriana) [inline buddi/sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-kang.ipynb:1]: just deleted, all good.

File: buddi/sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb
Comment (ivichadriana) [inline buddi/sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: just deleted, all good.

File: setup.py
Comment (ivichadriana) [inline setup.py:1]: cool, I've never seen this type of file

**Review by ivichadriana — COMMENTED** [review]
just want you to make sure you want the .gitignore file before merging. everything else looks good to me! BuDDi is a full grown model now 🥇

File: .gitignore
Comment (nrosed) [reply .gitignore:1]: yes, definitely want this file. It lets git know which files should not be added to git that may be automatically generated by the your code/environment/etc. In the future, you will likely need to generate this file too, it's very handy.

File: buddi/preprocessing/sc_preprocess.py (hunk 1 above)
Comment (nrosed) [reply buddi/preprocessing/sc_preprocess.py:57]: yeah -- I wonder why I never caught this before -__-

File: setup.py
Comment (nrosed) [reply setup.py:1]: when you make your first python package/method, this is a way for other people to easily install the same environment you used and all the things they need.

File: .gitignore
Comment (cgreene) [reply .gitignore:1]: This is a seriously comprehensive gitignore file! Is it a template from somewhere or something you've accumulated?

Comment (nrosed) [reply .gitignore:1]: haha, yes, it is comprehensive -- I did not accumulate, I got it from github "Python.gitignore": https://github.com/github/gitignore/blob/main/Python.gitignore

**Pull request merged**